pkg/detectors/uri/uri.go

```diff
@@ -23,7 +23,7 @@ var _ detectors.Detector = (*Scanner)(nil)
 var _ detectors.CustomFalsePositiveChecker = (*Scanner)(nil)
 
 var (
-	keyPat = regexp.MustCompile(`\b(?:https?:)?\/\/[\S]{3,50}:([\S]{3,50})@[-.%\w\/:]+\b`)
+	keyPat = regexp.MustCompile(`\b(?:https?:)?\/\/[\w-\.]{3,50}:([\w-\.]{3,50})@[-.%\w\/:]+\b`)
```

Comment:
`\S` matches any non-whitespace character, which is very broad. Instead, we are now using `\w`, which matches `[A-Za-z0-9_]`, and extending it by adding a few special characters to suit our needs.